scheduler/launchd.go

```diff
@@ -0,0 +1,149 @@
+	if err != nil {
+		return false, ErrModifyingLaunchd{origError: err}
+	}
+
+	launchdWasUnloaded, err = ld.disable(&fileFunctionsPassthrough{}, launchdAgentFilename)
+	if err != nil {
+		return launchdWasUnloaded, ErrModifyingLaunchd{origError: err}
+	}
+	return launchdWasUnloaded, nil
+}
+
+func (ld *launchd) Name() string {
+	return "launchd"
+}
+
+func (ld *launchd) enable(fileFunctions fileFunctionsInterface, launchdAgentFilename string) (
+	launchdFileWasCreated bool, err error) {
+
+	if _, err := fileFunctions.OsStat(launchdAgentFilename); os.IsNotExist(err) {
+		// file does not exist, write out default launchd agent file
+		if err = fileFunctions.IoutilWriteFile(
+			launchdAgentFilename, []byte(LaunchdFileContents), 0644); err != nil {
+			return false,
+				fmt.Errorf("%s didn't exist and failed to create it: %v", launchdAgentFilename, err)
+		}
+	}
+
+	_, err = ld.runLaunchctl("load", launchdAgentFilename)
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (ld *launchd) disable(fileFunctions fileFunctionsInterface, launchdAgentFilename string) (
+	launchdFileWasRemoved bool, err error) {
+
+	_, err = ld.runLaunchctl("unload", launchdAgentFilename)
+	if err != nil {
+		return false, err
+	}
+
+	if err := fileFunctions.OsRemove(launchdAgentFilename); err != nil {
+		return false, fmt.Errorf("failed to remove %s: %v", launchdAgentFilename, err)
+	}
+
+	return true, nil
+}
+
+func (ld *launchd) GetAgentFilename() (string, error) {
+	homeDirectory, err := homedir.Dir()
+	if err != nil {
+		return "", err
+	}
+
+	return path.Join(homeDirectory, LaunchAgentsSubdirectory, LaunchAgentFilename), nil
+}
+
+func (ld *launchd) runLaunchctl(verb string, filename string) (string, error) {
+	log.Printf("Running `%s %s %s`", launchctl, verb, filename)
+	cmd := exec.Command(launchctl, verb, filename)
+
+	out, err := cmd.CombinedOutput()
+
+	outString := string(out)
+	if err != nil {
+		return outString, err
+	}
+	if outString == filename+": No such file or directory\n" {
+		return outString, errCouldntFindLaunchdFile
+	}
+	return outString, nil
+}
+
+const launchctl string = "launchctl"
+
+// LaunchdFileContents is the agent file for running fk sync every 60 minutes
+const LaunchdFileContents = `<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+    <dict>
+        <key>Label</key>
+        <string>com.fluidkeys.sync</string>
+        <key>ProgramArguments</key>
+        <array>
+            <string>/usr/local/bin/fk</string>
+            <string>sync</string>
+            <string>--cron-output</string>
+        </array>
+        <key>StartInterval</key>
+        <integer>3600</integer>
+    </dict>
+</plist>
+`
+
+var (
+	// LaunchAgentsSubdirectory is the subdirectory for agent files, from the user's home directory
+	LaunchAgentsSubdirectory = path.Join("Library", "LaunchAgents")
```

Comment:
The comment says it's inside the user's *home directory*, but the code doesn't ...

The path `Library/LaunchAgents` is *not* equal to `/home/user/Library/LaunchAgents`... for a demo, try changing directory into `/` then running `fk` again...